src/main/java/com/gsheet/student/service/StudentService.java

```diff
@@ -95,7 +84,7 @@ public void deleteStudent(int id) {
             .data(List.of(resource))
             .build();
 
-        sheetRepository.updateValueRange(dto);
+        googleSheetService.updateValueRange(dto);
     }
 
     private int getNextStudentId() {
```

Comment:
Don't use this strange logic to build student id, use UUID for it. This logic contains a batch of problems that can be on future: multithreading, issues with student without first, middle and last names, issues related to line 103, you can delete last student and then create new one(new one have to be not with the previous student id)